components/Buttons/src/MDCButton.m

```diff
@@ -207,18 +213,20 @@ - (void)commonMDCButtonInit {
   _shouldRaiseOnTouch = YES;
   _uppercaseTitle = YES;
   _userElevations = [NSMutableDictionary dictionary];
-  _backgroundColors = [NSMutableDictionary dictionary];
   _accessibilityLabelForState = [NSMutableDictionary dictionary];
 
-  _backgroundColors[@(UIControlStateNormal)] = MDCColorFromRGB(MDCButtonDefaultBackgroundColor);
+  if (!_backgroundColors) {
+    // _backgroundColors may have already been initialized by setting the backgroundColor setter.
+    _backgroundColors = [NSMutableDictionary dictionary];
+    _backgroundColors[@(UIControlStateNormal)] = MDCColorFromRGB(MDCButtonDefaultBackgroundColor);
```

Comment:
[self updateBackgroundColor] is called at the end of both initializers.

We don't want to always write to the Normal backgroundColor because we might already have a value there, so we only write to it if the dictionary is uninitialized.